src/command.js

```diff
@@ -2,74 +2,115 @@ import program from 'commander';
+    const watchConfig = watchPath ? YAML.load(watchPath) : YAML.load(watchingConfigPath);
+    program
+      .version('0.1.0')
+      .option('-a, --addresses [n]', 'List of address', list, Commands.handelInputValues('ADDRESSES', watchConfig.addresses, ''))
+      .option('-f, --from [n]', 'From block', Commands.handelInputValues('FROM_BLOCK', watchConfig.from, 0))
+      .option('-t, --to [n]', 'To block', Commands.handelInputValues('TO_BLOCK', watchConfig.to, defaultBlockNumber))
+      .option('-q, --quick [n]', 'Quick Mode', Commands.handelInputValues('QUICK_MODE', watchConfig.quick, true))
+      .option('-s, --save-state [n]', 'Save state', Commands.handelInputValues('SAVE_STATE', watchConfig.saveState, null))
+      .option('-n, --node-url [n]', 'Node address', Commands.handelInputValues('RPC_URL', watchConfig.node, 'http://localhost:8545'))
+      .option('-l, --log-level [n]', 'Log level', Commands.handelInputValues('LOG_LEVEL', watchConfig.log_level, 'info'))
+      .option('-p, --socket-port [n]', 'IO socket port number', Commands.handelInputValues('SOCKET_IO_PORT', watchConfig.socket, 3030))
+      .option('-o,--output-type [n]', 'Output type', Commands.handelInputValues('OUTPUT_TYPE', watchConfig.outputType, 'terminal'))
+      .option('-e,--access-token [n]', 'etherscan accssess token', Commands.handelInputValues('ACCESS_TOKEN', watchConfig.accessToken, ''))
+      .parse(process.argv);
 
-  program.parse(process.argv);
-  if (typeof program === 'undefined') { throw new Error('No args are specifed in the command or in the .watch.yml file'); }
+    if (typeof program === 'undefined') { throw new Error('No args are specifed in the command or in the .watch.yml file'); }
+    // TODO: add validation to all parameters
+    this.to = Commands.validateBlock(program.to, lastBlockNumber);
+    this.from = Commands.validateBlock(program.from, program.to);
+    this.addresses = Commands.validateAddresses(program.addresses);
+    this.quickMode = program.quick;
+    this.lastBlockNumberFilePath =
+      program.saveState ? Commands.handleSaveSate(program.saveState, program.from) : program.from;
+    this.node = program.nodeUrl;
+    this.logLevel = program.logLevel;
+    this.socketPort = program.socketPort;
+    this.outputType = program.outputType;
+    this.accessToken = program.accessToken;
+    this.saveState =
+      program.saveState ? Commands.handleSaveSate(program.saveState, program.from) : program.from;
+  }
 
-  let addresses = null;
-  let from = program.from;
-  const to = program.to;
-  const quickMode = program.quick;
-  const saveStatePath = program.saveState;
-  let lastBlockNumberFilePath = null;
+  /**
+   * Handle input value priorities where ENV variable is higher in priority then config file
+   * then default value
+   * @param {string} envName
+   * @return value
+   */
+  static handelInputValues(envName, fileInst, defaultValue) {
+    return getEnv(envName) ? getEnv(envName) : fileInst || defaultValue;
```

Comment:
as far as I understand, I see you changed the priorities to be:
1- .env file
2- .watch
3- default

The questions is why should I put the address in .env? why there's possibility to put the watchConfig.node in watch file?

I think this will conduct to miss understand between .env and .watch.

Reply:
are you sure `handle` is good name? I would use `get` instead.

Reply:
By using commander the priority will be for 
1- CLI
2- ENV
3- .watch
4- Default value 

the reason for this is to take off the reliance on node as a server. We did a mistake assuming this application will be run from a server point of view. If you want a CLI tool + server tool then there is no running away from giving the user the ability to utilize anything be that a configuration file, env variable or CLI directly. 
In order to use the tool as fully CLI then we need to get rid of relying on Env variables. 

Regarding `handle` to `get` i will update now